src/order/well_founded.lean

```diff
@@ -113,6 +113,16 @@ begin
   rintro (hy | rfl), exact trans hy (wo.wf.lt_succ h), exact wo.wf.lt_succ h
 end
 
+section linear_order
+
+variables {β : Type*} [linear_order β] (h : well_founded ((<) : β → β → Prop))
+
+include h
+theorem self_le_of_strict_mono {φ : β → β} (hφ : strict_mono φ) : ∀ n, n ≤ φ n :=
+by { by_contra' h₁, have h₂ := h.min_mem _ h₁, exact h.not_lt_min _ h₁ (hφ h₂) h₂ }
```

Comment:
Not sure if you moved it just to remove it from the namespace, or whether you think this is a better location for it. You can also rename it to `_root_.well_founded.self_le_of_strict_mono` to make sure it doesn't live in the `function` namespace.

Reply:
The reason I'm putting it here is so that I can put it together with `eq_strict_mono_iff_eq_range`, from #11882.